src/Compiler/Facilities/BuildGraph.fs

```diff
@@ -274,36 +73,39 @@ type GraphNode<'T> private (computation: NodeCode<'T>, cachedResult: ValueOption
                                      ||| TaskContinuationOptions.NotOnFaulted
                                      ||| TaskContinuationOptions.ExecuteSynchronously)
                                 )
-                            |> NodeCode.AwaitTask
+                            |> Async.AwaitTask
+
+                        // Prevent deadlocks.
```

Comment:
Again, this is something I'd need help with. I understand that the idea is to run this synchronously whenever possible to get better callstacks, but I've observed this often deadlocks in CI tests.

Reply:
It deadlocked on the semaphore above, or somewhere else?

Reply:
It froze the CI tests IIRC, I don't think I ever got to repro this one locally, so this is kind of a black box to me.

Reply:
I removed this and I don't see anything wrong. Oh well.